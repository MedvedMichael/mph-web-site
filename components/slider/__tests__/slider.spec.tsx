import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Slider from '../slider'
import renderer from 'react-test-renderer'

test('should render', () => {
    const props = {
        images: [
            'some_url'
        ]
    }
    const res = shallow(<Slider {...props}/>)
    expect(toJson(res)).toMatchSnapshot()
})

test('should not render without images', () => {
    const props = {
        images: []
    }
    const res = shallow(<Slider {...props}/>)
    expect(toJson(res)).toMatchSnapshot()
})


test('should have no styles without bounds', () => {
    const bounds = {
        width: '100px', 
        height: '100px'
    }
    const props = {
        images: [
            'some_url'
        ],
        bounds
    }
    
    const res = renderer.create(<Slider {...props}/>).root.findAllByType('div')
    expect(res[0].props.style).toStrictEqual({...bounds})
})

