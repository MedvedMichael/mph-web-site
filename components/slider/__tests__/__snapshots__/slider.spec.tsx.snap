// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render without images 1`] = `""`;

exports[`should render 1`] = `
<styled.div
  style={Object {}}
>
  <ForwardRef(LoadableComponent)
    height="100%"
    images={
      Array [
        Object {
          "url": "some_url",
        },
      ]
    }
    width="100%"
  />
</styled.div>
`;
